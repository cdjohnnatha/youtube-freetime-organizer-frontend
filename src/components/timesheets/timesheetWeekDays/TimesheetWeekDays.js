import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Grid } from '@material-ui/core';

import TimesheetGridElements from '../timesheetGridElements/TimesheetGridElements';

const TimesheetWeekDays = ({ timesheetScheduleHours }) => {
  let renderTimesheetWeekDays = null;
  if (Array.isArray(timesheetScheduleHours) && timesheetScheduleHours.length > 0) {
    renderTimesheetWeekDays = timesheetScheduleHours.map(({ day_of_week, available_minutes }) => (
      <TimesheetGridElements
        title={day_of_week}
        content={`${available_minutes} min`}
        direction="column"
      />
    ));
  }

  return (
    <>
      <Typography component="h1" style={{ marginBottom: '1em' }}>
        Timesheet schedule:
      </Typography>
      <Grid container direction="row" justify="space-between">
        {renderTimesheetWeekDays}
      </Grid>
    </>
  );
};

TimesheetWeekDays.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
};

TimesheetWeekDays.defaultProps = {
  title: '',
  content: '',
}

export default TimesheetWeekDays;
