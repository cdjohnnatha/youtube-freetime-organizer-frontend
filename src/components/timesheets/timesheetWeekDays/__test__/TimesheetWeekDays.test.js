
import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import '../../../../../jest.setup';
import TimesheetWeekDays from '../TimesheetWeekDays';

import timesheetWeekDaysMock from './__mock__/timesheetWeekDays.json';
let wrapper = null;
let timesheetWeekDays = null;
describe('Test TimesheetWeekDays component', () => {
  test('TimesheetWeekDays snapshot renders', () => {
    const component = create(<TimesheetWeekDays timesheetScheduleHours={timesheetWeekDaysMock}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  beforeEach(() => {
    wrapper = mount(<TimesheetWeekDays timesheetScheduleHours={timesheetWeekDaysMock} />);
    timesheetWeekDays = wrapper.find('.timesheet_grid_elements');
  });
  test('TimesheetWeekDays component', () => {
    expect(wrapper.find('h1').at(0).text()).toBe('Timesheet schedule:');
    timesheetWeekDaysMock.forEach(({ day_of_week, available_minutes }, index) => {
      expect(wrapper.find('h1').at(index+1).text()).toBe(day_of_week);
      expect(wrapper.find('h6').at(index).text()).toBe(`${available_minutes} min`);
    })
  });
});