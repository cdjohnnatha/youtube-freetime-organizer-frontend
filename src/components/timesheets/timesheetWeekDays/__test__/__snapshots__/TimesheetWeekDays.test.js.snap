// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TimesheetWeekDays component TimesheetWeekDays snapshot renders 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-body1"
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    Timesheet schedule:
  </h1>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          MONDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          10 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          TUESDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          20 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          WEDNESDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          30 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          THURSDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          40 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          FRIDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          50 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          SATURDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          50 min
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      >
        <h1
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          SUNDAY
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          60 min
        </h6>
      </div>
    </div>
  </div>,
]
`;
